modules/asg/main.tf

```diff
@@ -0,0 +1,81 @@
+data "aws_ami" "amazon-linux-2" {
+  most_recent = true
+  owners      = ["amazon"]
+
+  filter {
+    name   = "name"
+    values = ["amzn2-ami-hvm-*-x86_64-ebs"]
+  }
+}
+
+resource "aws_launch_template" "t3micro-template" {
+  name_prefix   = "t3micro"
+  image_id      = "ami-0359b3157f016ae46"
+  instance_type = "t3.micro"
+  key_name                    = "MyKeyPair"
+  network_interfaces {
+    associate_public_ip_address = true
+  }
+
+
+  tags = {
+    Name = "1stasg"
```

Comment:
Tags and naming should be consistent across the entire project and meaningful.
Please think about which tags you could use and what would be the naming rules for resources in your project.
After that, let's have a discussion